### PR TITLE
botany activity generation from raw github events

### PR DESCRIPTION
**Pull request by andycmaj**
added initial raw github types. includes stuff that should appear in every github hook for the purposes of pulling it out in a typesafe way and normalizing

added structure and types for generating N botany activities from a raw github event.

here's the `botany_activity` published to that stream:

```
{ actor:
   { name: 'andycmaj',
     id: 1,
     created_at: null,
     primary_email: 'andycunn@gmail.com',
     users_data_sources: [ [Object] ],
     __typename: 'users' },
  type: 'code_review_created',
  source: 'github',
  data:
   { link: 'https://github.com/asynchronous-dev/async-www/pull/4',
     diffUrl: 'https://github.com/asynchronous-dev/async-www/pull/4.diff',
     additions: 2,
     deletions: 0,
     modifications: 1,
     commits: 1 } } +0ms
```

### CODE REVIEW

File: api/src/data/secretsStore.ts [hunk 1]
```diff
@@ -25,27 +27,21 @@ export const getDataSource = async (dataSource: string): Promise<Data_Sources['d
...
-  if (existingDataSource) {
-    await updateTeamDataSource(existingDataSource.id, data);
-  }
-  else {
-    await createTeamDataSource(data);
-  }
-};
-
-const getTeamDataSource = async (teamId: string, dataSourceId: string): Promise<Teams_Data_Sources> => {
+const getTeamDataSource = async (
```
Comment (davemeyer) [inline api/src/data/secretsStore.ts]: so i'm trying to figure out where this stuff belongs. a lot of the data access is going to be shared between the api and consumers. so should the entities and data access go in `/shared` or `/data` directory perhaps

File: ui/pages/index.tsx [hunk 2]
```diff
@@ -1,9 +1,12 @@
 import React from 'react';
 import styled from 'styled-components';
+import debug from 'shared/debug';
 
 const Title = styled.h1`
   font-size: 30px;
   color: ${({ theme }) => theme.colors.primary};
 `;
 
+debug('foo', 'bar');
```
Comment (davemeyer) [inline ui/pages/index.tsx]: :squint:
probably don't need that

File: api/src/data/secretsStore.ts (hunk 1 above)
Comment (andycmaj) [reply api/src/data/secretsStore.ts]: yeah i think it should go into shared and eventually we can refactor into a repository/service type pattern... i also have this: https://www.notion.so/asynchronous/reconcile-multiple-types-of-graphql-clients-ca64a11a7cef4d6b99046a16402466f3 so we can do that as part of that ticket.

File: ui/pages/index.tsx (hunk 2 above)
Comment (andycmaj) [reply ui/pages/index.tsx]: yarp! thx.

**Pull request merged**